api/controllers/NetworkController.js

```diff
@@ -1,3 +1,15 @@
+var updateCommunityIfModerator = (req, communityId, params, trx) =>
+  Membership.hasModeratorRole(req.session.userId, communityId)
+  .then(isModerator => {
```

Comment:
if you check Admin.isSignedIn first, you can skip the moderator role query entirely